app/services/true_layer/identify_applicant.rb

```diff
@@ -0,0 +1,53 @@
+module TrueLayer
+  class IdentifyApplicant
+    TRUE_LAYER_URL = 'https://api.truelayer.com'.freeze
+
+    attr_reader :token, :error
+
+    def initialize(token)
+      @token = token
+    end
+
+    def applicant
+      return login_error unless response.success?
+      Applicant.find_by(email: emails) || applicant_lookup_error
+    end
+
+    def emails
+      user_data[:emails]
+    end
+
+    def user_data
+      results[:results].first
+    end
+
+    def results
+      JSON.parse(response.body).deep_symbolize_keys
+    end
+
+    def response
+      @response ||= conn.get '/data/v1/info'.freeze
+    end
+
+    def conn
+      @conn ||= Faraday.new(url: TRUE_LAYER_URL) do |conn|
+        conn.authorization :Bearer, token
+        conn.response :logger if Rails.env.development?
```

Comment:
**[not a blocker/nit pick]** 💭 I would consider avoiding conditional code based on environment checks, and instead check some configuration that is setup to be enable for the environments you need, e.g.:

```ruby
...
conn.response :logger if debug?
...

def conf
  # either read from something that got set in the Rails app configuration or from an ENV variable (which is set to false by default unless for development)
  @conf ||= Rails.configuration.x.true_layer
end

def debug?
  !!conf.debug
end
```

That way we're not tied to a particular environment in the code and gives us flexibility to define when this options are enabled/disabled.